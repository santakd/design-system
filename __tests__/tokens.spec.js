// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/* eslint-env jest */
const path = require('path');
const fs = require('fs');
const _ = require('lodash');
const ROOT = path.resolve(__dirname, '../');
const BASE = path.resolve(ROOT, 'design-tokens');

function summarizeFilesInDirSync(directory) {
  return fs.readdirSync(directory);
}

function summarizeContentInFileSync(file) {
  return fs.readFileSync(file);
}

function getRawFile(file) {
  return summarizeContentInFileSync(path.resolve(BASE, file)).toString();
}

function compare(a, b) {
  return JSON.stringify(a).localeCompare(JSON.stringify(b));
}

function getTokenInfo(name) {
  const getFile = JSON.parse(getRawFile(`dist/${name}.raw.json`));
  return {
    dir: path.resolve(BASE, name),
    aliasDir: path.resolve(BASE, 'aliases'),
    raw: getFile,
    aliases: getFile.aliases,
    props: getFile.props
  };
}

const globalAccess = [
  'brandPrimary',
  'brandPrimaryActive',
  'brandPrimaryTransparent',
  'brandLight',
  'brandLightActive',
  'brandDark',
  'brandDarkActive',
  'brandAccessible',
  'brandAccessibleActive',
  'brandDisabled',
  'brandContrast',
  'brandContrastActive',
  'brandBackgroundPrimary',
  'brandBackgroundDark',
  'brandTextLink',
  'brandTextLinkActive',
  'colorBackground',
  'colorBackgroundAlt',
  'colorBackgroundAltInverse',
  'colorBackgroundRowHover',
  'colorBackgroundRowActive',
  'colorBackgroundRowSelected',
  'colorBackgroundRowNew',
  'colorBackgroundInverse',
  'colorBackgroundAnchor',
  'colorBackgroundBrowser',
  'colorBackgroundChromeMobile',
  'colorBackgroundChromeDesktop',
  'colorBackgroundHighlight',
  'colorBackgroundModal',
  'colorBackgroundModalBrand',
  'colorBackgroundNotificationBadge',
  'colorBackgroundNotificationBadgeHover',
  'colorBackgroundNotificationBadgeFocus',
  'colorBackgroundNotificationBadgeActive',
  'colorBackgroundNotificationNew',
  'colorBackgroundPayload',
  'colorBackgroundShade',
  'colorBackgroundStencil',
  'colorBackgroundStencilAlt',
  'colorBackgroundScrollbar',
  'colorBackgroundScrollbarTrack',
  'colorBrand',
  'colorBrandDark',
  'colorBackgroundButtonBrand',
  'colorBackgroundButtonBrandActive',
  'colorBackgroundButtonBrandHover',
  'colorBackgroundButtonBrandDisabled',
  'colorBackgroundButtonDefault',
  'colorBackgroundButtonDefaultHover',
  'colorBackgroundButtonDefaultFocus',
  'colorBackgroundButtonDefaultActive',
  'colorBackgroundButtonDefaultDisabled',
  'colorBackgroundButtonIcon',
  'colorBackgroundButtonIconHover',
  'colorBackgroundButtonIconFocus',
  'colorBackgroundButtonIconActive',
  'colorBackgroundButtonIconDisabled',
  'colorBackgroundButtonInverse',
  'colorBackgroundButtonInverseActive',
  'colorBackgroundButtonInverseDisabled',
  'colorBackgroundModalButton',
  'colorBackgroundModalButtonActive',
  'colorBackgroundInput',
  'colorBackgroundInputActive',
  'colorBackgroundInputCheckbox',
  'colorBackgroundInputCheckboxDisabled',
  'colorBackgroundInputCheckboxSelected',
  'colorBackgroundInputDisabled',
  'colorBackgroundInputError',
  'colorBackgroundPill',
  'colorBackgroundToast',
  'colorBackgroundToastSuccess',
  'colorBackgroundToastError',
  'colorTextBrand',
  'colorTextButtonBrand',
  'colorTextButtonBrandHover',
  'colorTextButtonBrandActive',
  'colorTextButtonBrandDisabled',
  'colorTextButtonDefault',
  'colorTextButtonDefaultHover',
  'colorTextButtonDefaultActive',
  'colorTextButtonDefaultDisabled',
  'colorTextButtonDefaultHint',
  'colorTextButtonInverse',
  'colorTextButtonInverseDisabled',
  'colorTextIconDefault',
  'colorTextIconDefaultHint',
  'colorTextIconDefaultHover',
  'colorTextIconDefaultActive',
  'colorTextIconDefaultDisabled',
  'colorTextIconInverse',
  'colorTextIconInverseHover',
  'colorTextIconInverseActive',
  'colorTextIconInverseDisabled',
  'colorTextLabel',
  'colorTextPlaceholder',
  'colorTextPlaceholderInverse',
  'colorTextRequired',
  'colorTextPill',
  'colorTextActionLabel',
  'colorTextActionLabelActive',
  'colorTextBrowser',
  'colorTextBrowserActive',
  'colorTextDefault',
  'colorTextError',
  'colorTextInputDisabled',
  'colorTextInputFocusInverse',
  'colorTextInputIcon',
  'colorTextInverse',
  'colorTextInverseWeak',
  'colorTextInverseActive',
  'colorTextInverseHover',
  'colorTextLink',
  'colorTextLinkActive',
  'colorTextLinkDisabled',
  'colorTextLinkFocus',
  'colorTextLinkHover',
  'colorTextLinkInverse',
  'colorTextLinkInverseHover',
  'colorTextLinkInverseActive',
  'colorTextLinkInverseDisabled',
  'colorTextModal',
  'colorTextModalButton',
  'colorTextStageLeft',
  'colorTextTabLabel',
  'colorTextTabLabelSelected',
  'colorTextTabLabelHover',
  'colorTextTabLabelFocus',
  'colorTextTabLabelActive',
  'colorTextTabLabelDisabled',
  'colorTextToast',
  'colorTextWeak',
  'colorTextIconBrand',
  'colorBorder',
  'colorBorderBrand',
  'colorBorderError',
  'colorBorderSuccess',
  'colorBorderWarning',
  'colorBorderTabSelected',
  'colorBorderSeparator',
  'colorBorderSeparatorAlt',
  'colorBorderSeparatorAlt',
  'colorBorderSeparatorInverse',
  'colorBorderRowSelected',
  'colorBorderRowSelectedHover',
  'colorBorderButtonBrand',
  'colorBorderButtonBrandDisabled',
  'colorBorderButtonDefault',
  'colorBorderButtonInverseDisabled',
  'colorBorderInput',
  'colorBorderInputActive',
  'colorBorderInputDisabled',
  'colorBorderInputCheckboxSelectedCheckmark',
  'fontFamily',
  'fontWeightLight',
  'fontWeightRegular',
  'fontWeightBold',
  'lineHeightHeading',
  'lineHeightText',
  'lineHeightReset',
  'lineHeightTab',
  'lineHeightButton',
  'lineHeightButtonSmall',
  'borderWidthThin',
  'borderWidthThick',
  'spacingXxxSmall',
  'spacingXxSmall',
  'spacingXSmall',
  'spacingSmall',
  'spacingMedium',
  'spacingLarge',
  'spacingXLarge',
  'spacingXxLarge',
  'varSpacingXxxSmall',
  'varSpacingXxSmall',
  'varSpacingXSmall',
  'varSpacingSmall',
  'varSpacingMedium',
  'varSpacingLarge',
  'varSpacingXLarge',
  'varSpacingXxLarge',
  'varSpacingHorizontalXxxSmall',
  'varSpacingHorizontalXxSmall',
  'varSpacingHorizontalXSmall',
  'varSpacingHorizontalSmall',
  'varSpacingHorizontalMedium',
  'varSpacingHorizontalLarge',
  'varSpacingHorizontalXLarge',
  'varSpacingHorizontalXxLarge',
  'varSpacingVerticalXxxSmall',
  'varSpacingVerticalXxSmall',
  'varSpacingVerticalXSmall',
  'varSpacingVerticalSmall',
  'varSpacingVerticalMedium',
  'varSpacingVerticalLarge',
  'varSpacingVerticalXLarge',
  'varSpacingVerticalXxLarge',
  'borderRadiusSmall',
  'borderRadiusMedium',
  'borderRadiusLarge',
  'borderRadiusPill',
  'borderRadiusCircle',
  'sizeXxSmall',
  'sizeXSmall',
  'sizeSmall',
  'sizeMedium',
  'sizeLarge',
  'sizeXLarge',
  'sizeXxLarge',
  'squareIconUtilitySmall',
  'squareIconUtilityMedium',
  'squareIconUtilityLarge',
  'squareIconLargeBoundary',
  'squareIconLargeBoundaryAlt',
  'squareIconLargeContent',
  'squareIconMediumBoundary',
  'squareIconMediumBoundaryAlt',
  'squareIconMediumContent',
  'squareIconSmallBoundary',
  'squareIconSmallContent',
  'squareIconXSmallBoundary',
  'squareIconXSmallContent',
  'shadowDrag',
  'shadowDropDown',
  'shadowHeader',
  'shadowButtonFocus',
  'shadowButtonFocusInverse',
  'durationInstantly',
  'durationImmediately',
  'durationQuickly',
  'durationPromptly',
  'durationSlowly',
  'durationPaused',
  'heightFooter',
  'heightHeader',
  'squareTappable',
  'widthStageLeftCollapsed',
  'widthStageLeftExpanded'
];

describe('design-tokens', () => {
  // Confirm primitive tokens were copied
  it('primitive tokens have been copied from @salesforce-ux/design-system-primitive-tokens', () => {
    expect(
      summarizeFilesInDirSync(getTokenInfo('primitive').dir)
    ).toMatchSnapshot();
  });

  // Confirm primitive aliases were copied
  it('primitive aliases have been copied from @salesforce-ux/design-system-primitive-tokens', () => {
    expect(
      summarizeFilesInDirSync(getTokenInfo('primitive').aliasDir)
    ).toMatchSnapshot();
  });

  // Confirm primitive aliases exist in ui-force
  it('primitive aliases are found and resolved in ui-force file', () => {
    const comparison = compare(
      getTokenInfo('primitive').aliases,
      getTokenInfo('ui-force').aliases
    );
    expect(comparison).toBe(0);
  });

  // Confirm primitive aliases exists in theme-one-salesforce
  it('primitive aliases are found in theme-one-salesforce file', () => {
    let namespaceArray = [];
    let primitiveArray = [];
    Object.keys(getTokenInfo('theme-one-salesforce').aliases).map(key => {
      namespaceArray.push(key);
    });
    Object.keys(getTokenInfo('primitive').aliases).map(key => {
      primitiveArray.push(key);
    });
    namespaceArray.forEach(key => {
      expect(namespaceArray).toContain(key);
    });
  });

  // Confirm primitive tokens are found in ui-force by filtering primitive metadata
  it('primitive tokens are found in ui-force file', () => {
    const rawFile = getTokenInfo('ui-force');
    let primitiveArray = [];
    Object.keys(rawFile.props).map(key => {
      const prop = rawFile.props[key];
      if (prop.primitive) {
        primitiveArray.push(prop.name);
      }
    });
    expect(primitiveArray).toMatchSnapshot();
  });

  // Check metadata so we don't cause platform regressions
  it('check that metadata for global access matches whats exposed on platform', () => {
    const rawFile = getTokenInfo('ui-force');
    let primitiveArray = [];
    Object.keys(rawFile.props).map(key => {
      const prop = rawFile.props[key];
      if (prop.access === 'global') {
        primitiveArray.push(_.camelCase(prop.name));
      }
    });
    globalAccess.map(key => {
      expect(primitiveArray).toContain(key);
    });
  });
});
