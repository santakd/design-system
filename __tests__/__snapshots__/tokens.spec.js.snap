// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design-tokens primitive aliases have been copied from @salesforce-ux/design-system-primitive-tokens 1`] = `
Array [
  "border-radius.yml",
  "border-width.yml",
  "color-palettes.yml",
  "colors.yml",
  "duration.yml",
  "font-family.yml",
  "font-size.yml",
  "font-weight.yml",
  "offset.yml",
  "sizing.yml",
  "spacing.yml",
]
`;

exports[`design-tokens primitive tokens are found in ui-force file 1`] = `
Array [
  "COLOR_BACKGROUND_ROW_ACTIVE",
  "COLOR_BACKGROUND_LIGHT",
  "COLOR_BACKGROUND_ROW_HOVER",
  "COLOR_BACKGROUND_ERROR_DARK",
  "COLOR_BACKGROUND_INFO",
  "COLOR_BACKGROUND_ALT_INVERSE",
  "COLOR_BACKGROUND_ROW_NEW",
  "COLOR_BACKGROUND_SUCCESS_DARK",
  "COLOR_BACKGROUND_STENCIL",
  "COLOR_BACKGROUND_WARNING",
  "COLOR_BACKGROUND_HIGHLIGHT_SEARCH",
  "COLOR_BACKGROUND_DESTRUCTIVE_HOVER",
  "COLOR_BACKGROUND",
  "COLOR_BRAND_DARK",
  "COLOR_BACKGROUND_BACKDROP",
  "COLOR_BACKGROUND_DARK",
  "COLOR_BRAND",
  "COLOR_BACKGROUND_DESTRUCTIVE",
  "COLOR_BACKGROUND_SUCCESS",
  "COLOR_BACKGROUND_WARNING_DARK",
  "COLOR_BACKGROUND_OFFLINE",
  "COLOR_BACKGROUND_STENCIL_ALT",
  "COLOR_BACKGROUND_INVERSE_LIGHT",
  "COLOR_BACKGROUND_CUSTOMER",
  "COLOR_BACKGROUND_ERROR",
  "COLOR_BACKGROUND_ALT",
  "COLOR_BACKGROUND_SUCCESS_DARKER",
  "COLOR_BACKGROUND_ROW_SELECTED",
  "COLOR_BRAND_DARKER",
  "COLOR_BACKGROUND_SELECTION",
  "COLOR_BACKGROUND_DESTRUCTIVE_ACTIVE",
  "COLOR_BACKGROUND_INVERSE",
  "COLOR_BACKGROUND_HIGHLIGHT",
  "COLOR_BORDER_ERROR_DARK",
  "COLOR_BORDER_BRAND_DARK",
  "COLOR_BORDER_INFO",
  "COLOR_BORDER_WARNING",
  "COLOR_BORDER_DESTRUCTIVE_HOVER",
  "COLOR_BORDER",
  "COLOR_BORDER_SUCCESS_DARK",
  "COLOR_BORDER_DESTRUCTIVE",
  "COLOR_BORDER_SEPARATOR_ALT_2",
  "COLOR_BORDER_OFFLINE",
  "COLOR_BORDER_SUCCESS",
  "COLOR_BORDER_SEPARATOR",
  "COLOR_BORDER_BRAND",
  "COLOR_BORDER_SELECTION_ACTIVE",
  "COLOR_BORDER_ERROR",
  "COLOR_BORDER_SELECTION_HOVER",
  "COLOR_BORDER_ROW_SELECTED_HOVER",
  "COLOR_BORDER_CUSTOMER",
  "COLOR_BORDER_SELECTION",
  "COLOR_BORDER_ROW_SELECTED",
  "COLOR_BORDER_SEPARATOR_ALT",
  "COLOR_BORDER_INVERSE",
  "COLOR_BORDER_DESTRUCTIVE_ACTIVE",
  "COLOR_BORDER_ERROR_ALT",
  "COLOR_BORDER_SEPARATOR_INVERSE",
  "BORDER_RADIUS_SMALL",
  "BORDER_RADIUS_MEDIUM",
  "BORDER_RADIUS_LARGE",
  "BORDER_RADIUS_CIRCLE",
  "BORDER_WIDTH_THIN",
  "BORDER_WIDTH_THICK",
  "SHADOW_ACTIVE",
  "SHADOW_DRAG",
  "SHADOW_DROP_DOWN",
  "DURATION_INSTANTLY",
  "DURATION_IMMEDIATELY",
  "DURATION_QUICKLY",
  "DURATION_PROMPTLY",
  "DURATION_SLOWLY",
  "DURATION_PAUSED",
  "FONT_FAMILY",
  "FONT_FAMILY_HEADING",
  "FONT_FAMILY_MONOSPACE",
  "FONT_SIZE_1",
  "FONT_SIZE_2",
  "FONT_SIZE_3",
  "VAR_FONT_SIZE_10",
  "FONT_SIZE_4",
  "VAR_FONT_SIZE_11",
  "FONT_SIZE_5",
  "FONT_SIZE_6",
  "FONT_SIZE_7",
  "FONT_SIZE_8",
  "VAR_FONT_SIZE_1",
  "FONT_SIZE_9",
  "VAR_FONT_SIZE_2",
  "VAR_FONT_SIZE_3",
  "FONT_SIZE_10",
  "VAR_FONT_SIZE_4",
  "FONT_SIZE_11",
  "VAR_FONT_SIZE_5",
  "VAR_FONT_SIZE_6",
  "VAR_FONT_SIZE_7",
  "VAR_FONT_SIZE_8",
  "VAR_FONT_SIZE_9",
  "FONT_WEIGHT_LIGHT",
  "FONT_WEIGHT_REGULAR",
  "FONT_WEIGHT_BOLD",
  "LINE_HEIGHT_HEADING",
  "LINE_HEIGHT_TEXT",
  "LINE_HEIGHT_RESET",
  "VAR_LINE_HEIGHT_TEXT",
  "OPACITY_5",
  "OPACITY_8",
  "BRAND_HEADER_CONTRAST_WEAK_DISABLED",
  "COLOR_GRAY_11",
  "BRAND_PRIMARY_TRANSPARENT_10",
  "COLOR_GRAY_12",
  "BRAND_BACKGROUND_DARK_TRANSPARENT",
  "COLOR_GRAY_13",
  "BRAND_HEADER",
  "BRAND_LIGHT_ACTIVE",
  "BRAND_HEADER_CONTRAST_WARM",
  "BRAND_HEADER_ICON_DISABLED",
  "BRAND_HEADER_CONTRAST_ACTIVE",
  "BRAND_BACKGROUND_PRIMARY",
  "BRAND_PRIMARY",
  "BRAND_HEADER_CONTRAST_WEAK_ACTIVE",
  "BRAND_CONTRAST_ACTIVE",
  "BRAND_HEADER_CONTRAST_COOL",
  "BRAND_HEADER_CONTRAST_INVERSE",
  "BRAND_DARK",
  "BRAND_ACCESSIBLE",
  "BRAND_BACKGROUND_DARK",
  "BRAND_TEXT_LINK_ACTIVE",
  "COLOR_GRAY_1",
  "COLOR_GRAY_2",
  "BRAND_HEADER_ICON_ACTIVE",
  "COLOR_GRAY_3",
  "BRAND_HEADER_ICON",
  "BRAND_DISABLED",
  "BRAND_TEXT_LINK",
  "COLOR_GRAY_4",
  "COLOR_GRAY_5",
  "COLOR_GRAY_6",
  "BRAND_ACCESSIBLE_ACTIVE",
  "COLOR_GRAY_7",
  "BRAND_DARK_ACTIVE",
  "COLOR_GRAY_8",
  "BRAND_HEADER_CONTRAST_INVERSE_ACTIVE",
  "BRAND_HEADER_CONTRAST_COOL_ACTIVE",
  "COLOR_GRAY_9",
  "BRAND_CONTRAST",
  "BRAND_HEADER_CONTRAST_WEAK",
  "BRAND_PRIMARY_ACTIVE",
  "BRAND_BACKGROUND_PRIMARY_TRANSPARENT",
  "BRAND_HEADER_CONTRAST",
  "BRAND_PRIMARY_TRANSPARENT",
  "BRAND_HEADER_CONTRAST_WARM_ACTIVE",
  "BRAND_PRIMARY_TRANSPARENT_40",
  "BRAND_LIGHT",
  "COLOR_GRAY_10",
  "SIZE_X_SMALL",
  "SIZE_XXX_SMALL",
  "SQUARE_ICON_UTILITY_LARGE",
  "SIZE_XX_SMALL",
  "SQUARE_ICON_SMALL_BOUNDARY",
  "SIZE_SMALL",
  "SQUARE_ICON_UTILITY_MEDIUM",
  "SQUARE_ICON_X_SMALL_BOUNDARY",
  "SQUARE_ICON_UTILITY_SMALL",
  "SQUARE_ICON_MEDIUM_BOUNDARY",
  "SQUARE_ICON_LARGE_CONTENT",
  "SQUARE_ICON_LARGE_BOUNDARY",
  "SIZE_MEDIUM",
  "SQUARE_ICON_MEDIUM_CONTENT",
  "SIZE_X_LARGE",
  "SQUARE_ICON_X_SMALL_CONTENT",
  "VAR_SQUARE_ICON_MEDIUM_BOUNDARY",
  "SQUARE_ICON_MEDIUM_BOUNDARY_ALT",
  "SIZE_XX_LARGE",
  "SQUARE_ICON_LARGE_BOUNDARY_ALT",
  "SIZE_LARGE",
  "SQUARE_ICON_SMALL_CONTENT",
  "SQUARE_ICON_XX_SMALL_BOUNDARY",
  "SQUARE_ICON_MEDIUM_CONTENT_ALT",
  "VAR_SPACING_X_LARGE",
  "VAR_SPACING_HORIZONTAL_SMALL",
  "VAR_SPACING_HORIZONTAL_XX_LARGE",
  "SPACING_X_SMALL",
  "SPACING_XXX_SMALL",
  "SPACING_XX_SMALL",
  "VAR_SPACING_VERTICAL_LARGE",
  "VAR_SPACING_LARGE",
  "VAR_SPACING_MEDIUM",
  "VAR_SPACING_VERTICAL_MEDIUM",
  "VAR_SPACING_VERTICAL_X_SMALL",
  "SPACING_SMALL",
  "VAR_SPACING_VERTICAL_XXX_SMALL",
  "VAR_SPACING_X_SMALL",
  "VAR_SPACING_XXX_SMALL",
  "VAR_SPACING_HORIZONTAL_X_LARGE",
  "VAR_SPACING_HORIZONTAL_XX_SMALL",
  "VAR_SPACING_XX_LARGE",
  "SPACING_MEDIUM",
  "VAR_SPACING_VERTICAL_XX_LARGE",
  "VAR_SPACING_VERTICAL_SMALL",
  "VAR_SPACING_SMALL",
  "SPACING_NONE",
  "VAR_SPACING_HORIZONTAL_LARGE",
  "SPACING_X_LARGE",
  "SPACING_XX_LARGE",
  "VAR_SPACING_HORIZONTAL_XXX_SMALL",
  "VAR_SPACING_HORIZONTAL_X_SMALL",
  "VAR_SPACING_XX_SMALL",
  "VAR_SPACING_HORIZONTAL_MEDIUM",
  "VAR_SPACING_VERTICAL_XX_SMALL",
  "VAR_SPACING_VERTICAL_X_LARGE",
  "SPACING_LARGE",
  "COLOR_TEXT_ACTION_LABEL",
  "COLOR_TEXT_LINK_INVERSE",
  "COLOR_TEXT_LINK_INVERSE_ACTIVE",
  "COLOR_TEXT_ACTION_LABEL_ACTIVE",
  "COLOR_TEXT_WARNING",
  "COLOR_TEXT_LINK_FOCUS",
  "COLOR_TEXT_DESTRUCTIVE_HOVER",
  "COLOR_TEXT_LINK_DISABLED",
  "COLOR_TEXT_DEFAULT",
  "COLOR_TEXT_DESTRUCTIVE",
  "COLOR_TEXT_LINK_HOVER",
  "COLOR_TEXT_SUCCESS",
  "COLOR_TEXT_WEAK",
  "COLOR_TEXT_PLACEHOLDER_INVERSE",
  "COLOR_TEXT_LINK",
  "COLOR_TEXT_WARNING_ALT",
  "COLOR_TEXT_ICON_DEFAULT",
  "COLOR_TEXT_BRAND",
  "COLOR_TEXT_ERROR",
  "COLOR_TEXT_CUSTOMER",
  "COLOR_TEXT_BRAND_PRIMARY",
  "COLOR_TEXT_LINK_ACTIVE",
  "COLOR_TEXT_REQUIRED",
  "COLOR_TEXT_LINK_INVERSE_DISABLED",
  "COLOR_TEXT_INVERSE",
  "COLOR_TEXT_PLACEHOLDER",
  "COLOR_TEXT_INVERSE_WEAK",
  "COLOR_TEXT_LINK_INVERSE_HOVER",
  "COLOR_TEXT_SUCCESS_INVERSE",
  "COLOR_TEXT_LABEL",
  "SQUARE_TAPPABLE",
  "SQUARE_TAPPABLE_SMALL",
  "SQUARE_TAPPABLE_X_SMALL",
  "SQUARE_TAPPABLE_XX_SMALL",
  "HEIGHT_TAPPABLE",
  "HEIGHT_TAPPABLE_SMALL",
  "Z_INDEX_DROPDOWN",
  "Z_INDEX_DOCKED",
  "Z_INDEX_REMINDER",
  "Z_INDEX_SPINNER",
  "Z_INDEX_DEFAULT",
  "Z_INDEX_DEEPDIVE",
  "Z_INDEX_TOAST",
  "Z_INDEX_DIALOG",
  "Z_INDEX_POPUP",
  "Z_INDEX_MODAL",
  "Z_INDEX_STICKY",
  "Z_INDEX_OVERLAY",
]
`;

exports[`design-tokens primitive tokens have been copied from @salesforce-ux/design-system-primitive-tokens 1`] = `
Array [
  "background-color.yml",
  "base.yml",
  "border-color.yml",
  "border-radius.yml",
  "border-width.yml",
  "box-shadow.yml",
  "duration.yml",
  "font-family.yml",
  "font-size.yml",
  "font-weight.yml",
  "line-height.yml",
  "opacity.yml",
  "palette-colors.yml",
  "sizing.yml",
  "spacing.yml",
  "text-color.yml",
  "touch.yml",
  "z-index.yml",
]
`;
